aws/resource_aws_api_gateway_documentation_version.go

```diff
@@ -0,0 +1,138 @@
+
+	restApiId := d.Get("rest_api_id").(string)
+
+	params := &apigateway.CreateDocumentationVersionInput{
+		DocumentationVersion: aws.String(d.Get("version").(string)),
+		RestApiId:            aws.String(restApiId),
+	}
+	if v, ok := d.GetOk("description"); ok {
+		params.Description = aws.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] Creating API Gateway Documentation Version: %s", params)
+
+	version, err := conn.CreateDocumentationVersion(params)
+	if err != nil {
+		return fmt.Errorf("Error creating API Gateway Documentation Version: %s", err)
+	}
+
+	d.SetId(restApiId + "/" + *version.Version)
+
+	return resourceAwsApiGatewayDocumentationVersionRead(d, meta)
+}
+
+func resourceAwsApiGatewayDocumentationVersionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+	log.Printf("[DEBUG] Reading API Gateway Documentation Version %s", d.Id())
+
+	apiId, docVersion, err := decodeApiGatewayDocVersionId(d.Id())
+	if err != nil {
+		return err
+	}
+
+	version, err := conn.GetDocumentationVersion(&apigateway.GetDocumentationVersionInput{
+		DocumentationVersion: aws.String(docVersion),
+		RestApiId:            aws.String(apiId),
+	})
+	if err != nil {
+		if isAWSErr(err, apigateway.ErrCodeNotFoundException, "") {
+			log.Printf("[WARN] API Gateway Documentation Version (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("version", docVersion)
+	d.Set("rest_api_id", apiId)
+	d.Set("description", version.Description)
+	d.Set("version", version.Version)
+
+	return nil
+}
+
+func resourceAwsApiGatewayDocumentationVersionUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+	log.Printf("[DEBUG] Updating API Gateway Documentation Version %s", d.Id())
+
+	_, err := conn.UpdateDocumentationVersion(&apigateway.UpdateDocumentationVersionInput{
+		DocumentationVersion: aws.String(d.Get("version").(string)),
+		RestApiId:            aws.String(d.Get("rest_api_id").(string)),
+		PatchOperations: []*apigateway.PatchOperation{
+			{
+				Op:    aws.String(apigateway.OpReplace),
+				Path:  aws.String("/description"),
+				Value: aws.String(d.Get("description").(string)),
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] Updated API Gateway Documentation Version %s", d.Id())
+
+	return resourceAwsApiGatewayDocumentationVersionRead(d, meta)
+}
+
+func resourceAwsApiGatewayDocumentationVersionDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+	log.Printf("[DEBUG] Deleting API Gateway Documentation Version: %s", d.Id())
+
+	_, err := conn.DeleteDocumentationVersion(&apigateway.DeleteDocumentationVersionInput{
+		DocumentationVersion: aws.String(d.Get("version").(string)),
+		RestApiId:            aws.String(d.Get("rest_api_id").(string)),
+	})
+	return err
+}
+
+func decodeApiGatewayDocVersionId(id string) (string, string, error) {
```

Comment:
Worst OCD nitpick maybe of all time 😢 : Can you spell out `Documentation` so we can keep everything consistent with `DocumentationVersion`